sky/serve/autoscalers.py

```diff
@@ -91,23 +141,16 @@ def update_version(self, version: int, spec: 'service_spec.SkyServiceSpec',
         self.min_replicas = spec.min_replicas
         self.max_replicas = (spec.max_replicas if spec.max_replicas is not None
                              else spec.min_replicas)
-        # Reclip self.target_num_replicas with new min and max replicas.
-        self.target_num_replicas = max(
-            self.min_replicas, min(self.max_replicas, self.target_num_replicas))
+        # Re-clip self.target_num_replicas with new min and max replicas.
+        self.target_num_replicas = self._clip_target_num_replicas(
+            self.target_num_replicas)
         self.update_mode = update_mode
 
     def collect_request_information(
             self, request_aggregator_info: Dict[str, Any]) -> None:
         """Collect request information from aggregator for autoscaling."""
         raise NotImplementedError
```

Comment:
For all the APIs a custom autoscaler needs to implement, we should keep them in the same section. We can try tot separate the utilities and APIs using:
```python
# --- APIs for custom autoscaler ---
def api_1():
  pass
def api_2():
  pass
# --- Utilities functions ---
def ...
```

Reply:
Good point! Done. PTAL again!